render/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/render/jpql/serializer/impl/JpqlCaseSerializer.kt

```diff
@@ -0,0 +1,34 @@
+package com.linecorp.kotlinjdsl.render.jpql.serializer.impl
+
+import com.linecorp.kotlinjdsl.querymodel.jpql.expression.impl.JpqlCase
+import com.linecorp.kotlinjdsl.render.RenderContext
+import com.linecorp.kotlinjdsl.render.jpql.serializer.JpqlRenderSerializer
+import com.linecorp.kotlinjdsl.render.jpql.serializer.JpqlSerializer
+import com.linecorp.kotlinjdsl.render.jpql.writer.JpqlWriter
+import kotlin.reflect.KClass
+
+class JpqlCaseSerializer : JpqlSerializer<JpqlCase<*>> {
+    override fun handledType(): KClass<JpqlCase<*>> {
+        return JpqlCase::class
+    }
+
+    override fun serialize(part: JpqlCase<*>, writer: JpqlWriter, context: RenderContext) {
+        val delegate = context.getValue(JpqlRenderSerializer)
+
+        writer.write("CASE")
+        writer.writeEach(part.whens.entries, separator = " ") {
```

Comment:
I think serializing the `CaseWhen` class to a String would be better done by creating a `JpqlCaseWhenSerializer` class, since the test method would be longer.

I don't think the `JpqlCaseSerializer` should necessarily know how to output `JpqlCaseWhen`.

Reply:
I have implemented JpqlCaseWhenSerializer and JpqlCaseWhen.

Reply:
I have rolled back for this review